src/latch_cli/services/k8s/develop.py

```diff
@@ -0,0 +1,283 @@
+from __future__ import annotations
+
+import asyncio
+import json
+import subprocess  # noqa: S404
+import sys
+from enum import Enum
+from typing import TYPE_CHECKING
+from urllib.parse import urljoin, urlparse
+
+import click
+import gql
+import websockets.client as websockets
+
+from latch.utils import current_workspace
+from latch_sdk_config.latch import NUCLEUS_URL
+from latch_sdk_gql.execute import execute
+
+from ...constants import docker_image_name_illegal_pat, latch_constants
+from ...utils import (
+    TemporarySSHCredentials,
+    get_auth_header,
+    identifier_suffix_from_str,
+)
+from .ws_utils import forward_stdio
+
+if TYPE_CHECKING:
+    from pathlib import Path
+
+max_polls = 1800
+
+
+# not going to allow most of these for now
+class InstanceSize(str, Enum):
+    small_task = "small_task"
+    medium_task = "medium_task"
+    # large_task = "large_task"
+    small_gpu_task = "small_gpu_task"
+    # large_gpu_task = "large_gpu_task"
+    # v100_x1_task = "v100_x1_task"
+    # v100_x4_task = "v100_x4_task"
+    # v100_x8_task = "v100_x8_task"
+    # g6e_xlarge_task = "g6e_xlarge_task"
+    # g6e_2xlarge_task = "g6e_2xlarge_task"
+    # g6e_4xlarge_task = "g6e_4xlarge_task"
+    # g6e_8xlarge_task = "g6e_8xlarge_task"
+    # g6e_12xlarge_task = "g6e_12xlarge_task"
+    # g6e_16xlarge_task = "g6e_16xlarge_task"
+    # g6e_24xlarge_task = "g6e_24xlarge_task"
+    # g6e_48xlarge_task = "g6e_48xlarge_task"
+
+
+def workflow_name(pkg_root: Path) -> str:
+    name_path = pkg_root / latch_constants.pkg_workflow_name
+    if not name_path.exists():
+        click.secho(
+            "Unable to parse workflow name - please make sure you have registered your workflow first.",
+            fg="red",
+        )
+        raise click.exceptions.Exit(1)
+
+    return name_path.read_text().strip()
+
+
+def get_image(pkg_root: Path) -> str:
+    ws_id = current_workspace()
+    wf_name = workflow_name(pkg_root)
+
+    prefix = ws_id
+    if int(ws_id) < 10:
+        prefix = f"x{ws_id}"
+
+    suffix = wf_name
+    suffix = identifier_suffix_from_str(wf_name).lower()
+    suffix = docker_image_name_illegal_pat.sub("_", wf_name)
+
+    return f"{prefix}_{suffix}"
+
+
+def get_latest_registered_version(pkg_root: Path) -> str:
```

Comment:
You could inline all of these? IDRC but seems like they are one use and might be easier to read.